### DISCUSSION TITLE
to use OpenMP, do i need to have OpenMP insaled or just by having OpenMP as feature is enough?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#1888] [General]
OpenMP is something inside ImageMagick or i need to install OpenMP from the repository too?

**Comment by dlemstra** [comment]
Which OS are you on?

**Reply by ghost** [reply]
Arch Linux

**Comment by urban-warrior** [comment]
OpenMP is typically a delegate library or part of the compiler build chain.  It is not part of the ImageMagick distribution other than it is enabled if detected @ ImageMagick build time.

**Reply by ghost** [reply]
so i do need to install OpenMP right?

**Reply by urban-warrior** [reply]
Depends on the compiler.  Some compilers have OpenMP built-in.  Others depend on libgomp.  If all goes well with the ImageMagick build, you can verify OpenMP support by checking the build features:

```
convert --version | grep Features
Features: Cipher DPC HDRI Modules OpenMP(4.5) TCMalloc
```

**Reply by ghost** [reply]
yes i have the feature
so ImageMagick is already using it?

**Reply by urban-warrior** [reply]
If OpenMP is an enabled feature, ImageMagick will use it wherever there is support for it... mostly the image processing algorithms.  Reading and writing of an image is still single threaded.
